### ISSUE TITLE
Nix flake show: error: a 'x86_64-darwin' with features {} is required to build ...-emanote.drv'

### ISSUE THREAD

**Issue by j-steinbach** [srid/emanote#132]
**Describe the bug**

When trying to either show or include the flake in my NixOS system, I get the following error:

`error: a 'x86_64-darwin' with features {} is required to build '/nix/store/asl9s2l82lxah6q5vx2n476zq5s8s3h9-cabal2nix-emanote.drv', but I am a 'x86_64-linux' with features {benchmark, big-parallel, kvm, nixos-test}.`

**To Reproduce**

Run `nix flake show "github:srid/emanote"`on a x86_64-linux system.

**Expected behavior**

I expect the flake outputs to be shown, similar to [nix flake show](https://nixos.org/manual/nix/unstable/command-ref/new-cli/nix3-flake-show.html). 

I also expect the system to build when I include the flake in my flake-inputs (`emanote.url = "github:srid/emanote";`) and adding `inputs.emanote` to my modules. (Might be doing something wrong here, so better focus on `flake show`)

***Note*** I am not the most advanced Nix user, so this might also be something on my end. 

**Desktop (please complete the following information):**

`nix-shell -p nix-info --run "nix-info -m"`:

 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.64, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210908_3c56f62`
 - channels(root): `"nixos-21.05.3021.8b0b81dab17"`
 - channels(jst): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by srid** [comment]
Hmm, fwiw I have no issues using this flake on my nixos-config:

https://github.com/srid/nixos-config/blob/f70e13b40a88f2d5fbbf0dcb43d8b5db40271966/flake.nix#L18

I simply refer to it as `inputs.emanote.outputs.defaultPackage.${system};` (is this not how one refers to the flake?)

https://github.com/srid/nixos-config/blob/f70e13b40a88f2d5fbbf0dcb43d8b5db40271966/home.nix#L6

I don't know what the `nix flake show` error is all about, or what to be done to address it.

**Comment by j-steinbach** [comment]
> I don't know what the nix flake show error is all about, or what to be done to address it.

Haha, guess that makes us two :+1: 

> (is this not how one refers to the flake?)

Actually I have no idea. I just took a look at the Neuron and Emanote documents and the Nixos Wiki but couldn't find a clear, understandable answer -- so I went with my gut. This is also why I mentioned the `nix flake show` error first -- that one is independent of my config. 

> Hmm, fwiw I have no issues using this flake on my nixos-config:

As I am currently starting at a few other dumpster fires in my own config (you really don't want to know) I am going to extinguish them first (or die trying), then I take a look at your config and try to get it to work myself and then --if that doesn't work-- I come back and bug you about it again. 

That means I will close this for the time being -- if worst comes to worst, we can open it again.


-----

PS: Emanote looks absolutely awesome!
